### ISSUE TITLE
Allow creating an API key through config/env var

### ISSUE THREAD

**Issue by trehn** [erlio/vernemq#643]
I want other containers to talk to VerneMQ using the HTTP API. To do that, I need an API key. The current interface to create API keys is running `vmq-admin api-key create`. This is difficult to automate in containers (and config management), because it needs to run *once* inside the VerneMQ container and the result must be made available to the other containers (that will act as API clients).

It would be a lot easier if there was a config directive that allows me to set an API key in advance. Since all config directives can be set as env vars, configuration via Kubernets or Docker Compose would be easy.

**Comment by ioolkos** [comment]
@trehn thanks for your proposition!
It makes sense to me. But I'm easily convinced.

I wonder what @larshesel and @dergraf will say :)

**Comment by larshesel** [comment]
There's the `vmq-admin api-key add` which allows you to set a specific key. Not sure if that's exposed so it can be set via the env yet, though, but definitely less complex to work with than `vmq-admin api-key create`.

**Comment by dergraf** [comment]
@trehn what would be your suggestion as the `vmq-admin` allows you to generate multiple different API keys. Would a single API key be enough, conffigured via some `VERNEMQ_HTTP_API_KEY` or similar?

**Comment by trehn** [comment]
@dergraf Yes, a single API key would be sufficient IMO. If that key ever became compromised, one would probably just change it and redeploy all containers using that key rather than just changing a single key specific to one service.

**Comment by dergraf** [comment]
@trehn would you like to contribute a PR in erlio/docker-vernemq?

**Comment by kdaveid** [comment]
The described use case is valid - we have the same problems with deploying our VerneMQ brokers with docker. We would like to inject an api-key during startup so that we can monitor via '/health' on the api. Another solution (for our use case) would be to open '/health' requests to non-authenticated requests.

**Comment by larshesel** [comment]
The `/health` endpoint is open by default. The API key mentioned here is for the HTTP management API, see https://docs.vernemq.com/live-administration/http-administration

**Comment by lnovara** [comment]
Any news about this issue and the associated PRs? I think that the use case is very relevant.

Thank you.

**Comment by estin** [comment]
Set api key by plugin initialization works for me.
May be it would help some one.
Something like this

```erlang
-define(ENV_API_KEYS, http_mgmt_api_keys).
...

NewKey = list_to_binary(os:getenv("MY_API_KEY")),
Keys = vmq_config:get_env(vmq_server, ?ENV_API_KEYS, []),
Keys1 = lists:delete(NewKey, Keys),
vmq_config:set_global_env(vmq_server, ?ENV_API_KEYS, [NewKey|Keys1], true),
 ```

**Comment by rebsp** [comment]
@estin could you give a bit more context to that? I'm not familiar with erlang and would like to achieve the same as OP

**Comment by estin** [comment]
@trehn special for you )
 https://github.com/estin/vernemq_demo_plugin

This solution without ACL. Some more code needs to enable ACL

**Comment by rebsp** [comment]
Thanks @estin !
